### DISCUSSION TITLE
move p outside the body

### DISCUSSION THREAD

**Discussion by APerricone** [cssbattle/forum#261] [Q&A] [answered]
Hello, I was golfing css, and I saw that some solution uses p outside the body without the position fixed or something else. For example:
```html
<p><style>*{background:red}*+*{background:blue;margin:100;border:1px solid white}p{height:40;background:yellow;margin:-20+20}
```
produces:
![image](https://user-images.githubusercontent.com/22457168/168522623-5005ddea-06a4-41b2-986e-aaffa0b104c3.png)
but I remove the border it produces this completly different result:
![image](https://user-images.githubusercontent.com/22457168/168522694-db11bbaf-dc68-42e0-94cd-ef68085e3b23.png)
It is a correct behaviour? somebody knows different method to allow the p outside the body that does not change the look of the body (ie without adding border)

**Comment by APerricone** [comment]
looks like it is a well knowed behaviour, see https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing

**Comment by SelenIT** [comment, marked as answer]
As @APerricone said, it's the specified behavior. It was designed that way because CSS was created to style text documents in first place. Some ways to get rid of it include:

- adding `padding` to the parent
- adding `border` to the parent (lucky case if you use it anyway)
- adding at least one character of text (a non-breaking space `&nbsp;` works fine) _before_ the `<p>` and increasing the negative margin accordingly
- using the `<img>` element instead of `<p>` (it's a replaced inline element, basically working as `inline-block`, and its margins never collapse)
